### DISCUSSION TITLE
Reflection from BusinessBase

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#2555] [Questions]
Rocky,

I am wondering if it's a good idea to call the factory class via reflection from the businesBase Class?  Right now I use an ObjectFactory with the [Csla.Server.ObjectFactory(typeof(CompanyParameterFactory), "Create", "GetAsync", "UpdateAsync", "DeleteAsync")].  

Could I do something like:

``

     [Serializable]      
     public class CompanyParameterInfoPersist : InjectableBusinessBase<CompanyParameterInfoPersist>, ICompanyParameterInfoPersist
     {
     [Fetch]
     private Async Task<CompanyInfo> GetAsync(int companyId)
       {
         _methodName = "GetAsync";

         dynamic result = ExecuteWithReflectionFactory.ExecuteWithReflection(_assemblyName, _instanceName, _methodName, new object[] { companyId });

         CompanyInfo task1 = await result;

         return MapItem(task1);
        }
    }
``

This would allow me to reduce the number and size of the dlls distributed to the wpf front end.

What do you think?

**Comment by rockfordlhotka** [comment]
> This would allow me to reduce the number and size of the dlls distributed to the wpf front end.

These are not related issues.

Assuming you are using a remote data portal, you only need to deploy the DAL assemblies to the server. 

Just because the business assembly _references_ a DLL doesn't mean you need to deploy it. If that DLL is never invoked by the client-side code (and it sure shouldn't be!!), then .NET will never try to load that DLL and so it doesn't matter that it wasn't deployed.

So I'd recommend actually referencing the DAL assembly, injecting it into the data portal operation methods via DI, and just not deploying it to the client.
